### ISSUE TITLE
Crash Bandicoot 4 doesn't launch.

### ISSUE THREAD

**Issue by Ethan-BB** [dafzor/bnetlauncher#57]
With version 2.14, trying to start a game with the "bnetlauncher.exe cb4" command only opens the battle.net window, it doesn't select the correct game and doesn't start anything, I need to go to the correct tab and click on the play button after that overlay works fine.

I tried sending some commands directly:
"Battle.net.exe --game=wlby" opens the correct game tab, just doesn't start the game (as expected)
"Battle.net.exe --exec=wlby" doesn't work

**Comment by dafzor** [comment]
Hi, are you still having the issue?

**Comment by Ethan-BB** [comment]
Yeah, same behaviour on v2.15
Not sure if it makes a difference, I have a Friends List (old one) always open on the desktop.

**Comment by dafzor** [comment]
The battle.net command is actually `Battle.net.exe --exec "launch wlby"` so make sure that works.

If bettle.net client is working, please enable loging in bnetlauncher (see readme) and post a copy of the log.

**Comment by Ethan-BB** [comment]
Yeah, that command doesn't work. Just opens Battlenet on the last open game.

**Comment by dafzor** [comment]
> Yeah, that command doesn't work. Just opens Battlenet on the last open game.

It's possible blizzard changed the code for the game or broke the launch functionality for it. 

Can you confirm it works for other games? You can see the [list here](https://github.com/dafzor/bnetlauncher/blob/master/bnetlauncher/Resources/gamesdb.ini) for the cmd (pick one of the battlenet games, not battlenet2)?

**Comment by Ethan-BB** [comment]
StarCraft: Remastered, StarCraft 2, WarCraft 3: Reforged, Hearthstone, Overwatch and World of Warcraft all work fine.
Is there a way I could capture the command used when launching a game from the context menu from the tray? Because from there it starts fine.

**Comment by dafzor** [comment]
The launch code can be extracted from `products.db` which is an hidden file in `C:\ProgramData\Battle.net\Agent`. Opening with notepad should allow you to see some text information like the one bellow:
```
diablo3d3D
D:/Games/Diablo IIIeu
```

Where `diablo3` is the game `d3` the launch code and `D:/Games/Diablo III` the install path. Unfortunately the launch code is case sensitive and not show in the correct case. In Diablo 3 case the code is `D3` and not `d3`.

**Comment by Ethan-BB** [comment]
I can confirm `wlby` is correct, however I have a feeling `--exec "launch wlby"` isn't.
But I have no clue how to get the command used, Window Spy from HotkeyP is not working on Battlenet.

**Comment by dafzor** [comment]
You can try changing the code to all caps, or just first uppercase letter and see if that makes any difference.

Beyond that you could try adding an entry using "battlenet2" which uses different codes or try to reach out to the devs on blizzard forums so they can fix the client (assuming it's a bug on their end).

**Comment by Ethan-BB** [comment]
Finally had time to test this, `--exec "launch WLBY"` (all capitals) **will** launch the game correctly.
However, putting it in all capitals into gamesdb.ini is not working for some reason ... maybe parsing is ignoring the letter case?

**Comment by dafzor** [comment]
Try creating a new gamedb.ini entry instead of overwriting the setting on the existing one.

**Comment by Ethan-BB** [comment]
Yeah, that works.
Thank you!
